### PR TITLE
JSAB2-49 Battle frontend

### PR DESCRIPTION
**Pull request by kyya**

### CODE REVIEW

**Review by HanqingZ — APPROVED** [review]

**Review by HONGTAO-T — APPROVED** [review]

**Review by GuJialu — APPROVED** [review]

File: frontend/src/Battle/index.jsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  const [active, setActive] = useState(null);
+  const [players] = useState(mockedUsers);
+
+  function handleBattleStart(id, name) {
+    Toast.show({
+      text: `You are going to battle with ${name.toUpperCase()}`,
+    });
+  }
+
+  // useEffect(() => {
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx]: this is for connecting the backend right?

File: frontend/src/Battle/index.jsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  //   if (active) {
+  //     fetch(`http://${SERVER_URL}/battle/planet/${active}`)
+  //       .then((response) => {
+  //         if (response.status === 200) {
+  //           return response.json();
+  //         }
+  //         throw new Error('Unexpected status code');
+  //       })
+  //       .then((response) => setPlayers(response.usersInPlanet))
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx]:
```suggestion
  //       .then((response) => setPlayers(response.usersOnPlanet))
```

File: frontend/src/Battle/index.jsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  //         });
+  //       });
+  //   }
+  // }, [active]);
+
+  function renderPlayers() {
+    if (!players.length) {
+      return (
+        <CardView>
+          <Text>No players in this planet.</Text>
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx]:
```suggestion
          <Text>No players on this planet.</Text>
```

File: frontend/src/Battle/index.jsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        </CardView>
+      );
+    }
+    return (
+      <View>
+        {
+          players.map(({
+            _id: key, username: name, hp, attack, defence, planetList, allLevels,
+          }) => (
+            <PlayerItem
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx:96]: we always called them users, why change now to player?

**Review by evasimonyi — CHANGES_REQUESTED** [review]
the second user cannot fit here, and the user container is even smaller on iphone
I suggest making 3 lines instead of 2 for the user data in the container
and in the first line it could be the username, and without the pink container, only the name

Screenshot_2020-01-15-11-37-22-458_host exp exponent

File: frontend/src/Battle/index.jsx (hunk 1 above)
Comment (kyya) [reply frontend/src/Battle/index.jsx]: Yes, just waiting for the backend to be merged and I can start the connect work quickly.

File: frontend/src/Battle/index.jsx (hunk 4 above)
Comment (kyya) [reply frontend/src/Battle/index.jsx:96]: I think we named users in common apps like shopping app.
and we name players in a game app.

**Review by GuJialu — APPROVED** [review]

File: frontend/src/Battle/index.jsx (hunk 3 above)
Comment (evasimonyi) [reply frontend/src/Battle/index.jsx]: please change!

File: frontend/src/Battle/index.jsx (hunk 2 above)
Comment (evasimonyi) [reply frontend/src/Battle/index.jsx]: please change to `on`!

don't leave here commented out code!

File: frontend/src/Battle/index.jsx (hunk 1 above)
Comment (evasimonyi) [reply frontend/src/Battle/index.jsx]: battle map backend is merged

File: frontend/src/Battle/index.jsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
+import React, { useState } from 'react';
+import {
+  View, StyleSheet, ScrollView, Text, Alert, Button,
+} from 'react-native';
+// import { SERVER_URL } from 'react-native-dotenv';
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx]: delete comments please

File: frontend/src/Battle/index.jsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    height: 64,
+  },
+  scrollView: {
+    flex: 1,
+  },
+});
+
+const PLANETS = ['blue', 'green', 'yellow', 'red', 'purple'];
+
+const mockedUsers = [
```
Comment (evasimonyi) [inline frontend/src/Battle/index.jsx]: battle map backend is already merged, can you please use real data on this branch?
if not, what is it depending on?

**Review by evasimonyi — CHANGES_REQUESTED** [review]

File: frontend/src/Battle/index.jsx (hunk 6 above)
Comment (kyya) [reply frontend/src/Battle/index.jsx]: Of course, I can connect on this branch.

**Review by HONGTAO-T — APPROVED** [review]

**Review by HONGTAO-T — APPROVED** [review]

**Review by GuJialu — APPROVED** [review]

**Review by evasimonyi — APPROVED** [review]

**Pull request merged**